### PR TITLE
fix(notification): create only one notification per user

### PR DESCRIPTION
**Pull request by Phil91**
the notifications will only be created once per request for each user and type

Refs: CPLP-2526

### CODE REVIEW

File: src/marketplace/Offers.Library/Service/OfferSubscriptionService.cs [hunk 1]
```diff
@@ -261,6 +261,11 @@ await _offerSetupService
         
         await foreach (var receiver in _portalRepositories.GetInstance<IUserRepository>().GetServiceProviderCompanyUserWithRoleIdAsync(offerId, roleData))
         {
+            if (!offerProviderDetails.SalesManagerId.HasValue || receiver != offerProviderDetails.SalesManagerId.Value)
```
Comment (ntruchsess) [inline src/marketplace/Offers.Library/Service/OfferSubscriptionService.cs:264]: I'm pretty sure the second condition is incorrect. As a first notification for the salesmanager is being created in line 244 to skip the creation of a second notification for the same salesManager it must be 'receiver **==** offerProvierDetails.SalesManagerId.Value'.

File: src/registration/ApplicationActivation.Library/ApplicationActivationService.cs [hunk 2]
```diff
@@ -105,7 +105,7 @@ private async Task<IApplicationChecklistService.WorkerChecklistProcessStepExecut
         });
 
         var notifications = _settings.WelcomeNotificationTypeIds.Select(x => (default(string), x));
-        await _notificationService.CreateNotifications(_settings.CompanyAdminRoles, null, notifications, companyId).ConfigureAwait(false);
+        await _notificationService.CreateNotifications(_settings.CompanyAdminRoles, null, notifications, companyId).ToListAsync().ConfigureAwait(false);
```
Comment (ntruchsess) [inline src/registration/ApplicationActivation.Library/ApplicationActivationService.cs:108]: the respective test (ApplicationActivationTests.cs) needs to be adjusted as well. It contains a fake implementation using 'Invoke', while it should call ReturnsLazily returning IAsnycEnumerable.

File: src/notifications/Notifications.Library/NotificationService.cs [hunk 3]
```diff
@@ -53,6 +53,7 @@ public NotificationService(IPortalRepositories portalRepositories)
         await foreach (var receiver in _portalRepositories.GetInstance<IUserRepository>().GetCompanyUserWithRoleIdForCompany(roleData, companyId))
         {
             CreateNotification(receiver, creatorId, notifications, notificationRepository);
+            yield return receiver;
```
Comment (ntruchsess) [inline src/notifications/Notifications.Library/NotificationService.cs:56]: The unit-tests somehow should not just test that the CreateNotifications-method is being called, but also test that the IAsyncEnumerable which is being returned is a actually being iterated.

**Review by ntruchsess — CHANGES_REQUESTED** [review]

File: src/marketplace/Offers.Library/Service/OfferSubscriptionService.cs (hunk 1 above)
Comment (Phil91) [reply src/marketplace/Offers.Library/Service/OfferSubscriptionService.cs]: absolutly right. i've corrected the check and extended the unit tests for it.

File: src/registration/ApplicationActivation.Library/ApplicationActivationService.cs (hunk 2 above)
Comment (Phil91) [reply src/registration/ApplicationActivation.Library/ApplicationActivationService.cs:108]: added the return to the unitTest

**Review by ntruchsess — APPROVED** [review]

**Pull request merged**